src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.cs

```diff
@@ -0,0 +1,704 @@
+                {
+                    // STEP 3: If expression is "@(x)" copy specified list with its metadata, otherwise just treat as string
+                    bool isItemListExpression;
+                    ProcessSingleItemVectorExpressionForInclude(includeSplitEscaped, operationBuilder, itemElement.IncludeLocation, out isItemListExpression);
+
+                    if (!isItemListExpression)
+                    {
+                        // The expression is not of the form "@(X)". Treat as string
+
+                        //  Code corresponds to EngineFileUtilities.GetFileList
+                        bool containsEscapedWildcards = EscapingUtilities.ContainsEscapedWildcards(includeSplitEscaped);
+                        bool containsRealWildcards = FileMatcher.HasWildcards(includeSplitEscaped);
+
+                        if (containsEscapedWildcards && containsRealWildcards)
+                        {
+                            // Umm, this makes no sense.  The item's Include has both escaped wildcards and 
+                            // real wildcards.  What does he want us to do?  Go to the file system and find
+                            // files that literally have '*' in their filename?  Well, that's not going to 
+                            // happen because '*' is an illegal character to have in a filename.
+
+                            // Just return the original string.
+                            operationBuilder.Operations.Add(Tuple.Create(IncludeOperationType.Value, (object) includeSplitEscaped));
+                        }
+                        else if (!containsEscapedWildcards && containsRealWildcards)
+                        {
+                            // Unescape before handing it to the filesystem.
+                            string filespecUnescaped = EscapingUtilities.UnescapeAll(includeSplitEscaped);
+                            operationBuilder.Operations.Add(Tuple.Create(IncludeOperationType.Glob, (object)filespecUnescaped));
+                        }
+                        else
+                        {
+                            // No real wildcards means we just return the original string.  Don't even bother 
+                            // escaping ... it should already be escaped appropriately since it came directly
+                            // from the project file
+                            operationBuilder.Operations.Add(Tuple.Create(IncludeOperationType.Value, (object) includeSplitEscaped));
+                        }
+
+                    }
+                }
+            }
+
+            //  Code corresponds to Evaluator.EvaluateItemElement
+
+            // STEP 4: Evaluate, split, expand and subtract any Exclude
+            if (itemElement.Exclude.Length > 0)
+            {
+                string evaluatedExclude = _expander.ExpandIntoStringLeaveEscaped(itemElement.Exclude, ExpanderOptions.ExpandProperties, itemElement.ExcludeLocation);
```

Comment:
>itemElement.Exclude [](start = 81, length = 19)

For GetItemProvenance, the exclude string will probably have to go through the same treatment as the include, to detect values, globs, expressions, etc